pkg/config/config.go

```diff
@@ -1277,6 +1289,27 @@ func (c *RuntimeConfig) initializeRuntimeFeatures() {
 		if err := json.Unmarshal(output, &handler.features); err != nil {
 			logrus.Errorf("Unmarshalling OCI features failed: %s", err)
 		}
+
+		if handler.RuntimeSupportsIDMap() {
+			logrus.Debugf("Runtime handler %q supports User and Group ID-mappings", name)
+		}
+
+		// Recursive Read-only (RRO) mounts require runtime handler support,
+		// such as runc v1.1 or crun v1.4. For Linux, the minimal kernel
+		// version 5.12 or a kernel with the necessary changes backported
+		// are required.
+		rro := handler.RuntimeSupportsMountFlag("rro")
+		if rro {
+			logrus.Debugf("Runtime handler %q supports Recursive Read-only (RRO) mounts", name)
+
+			// A given runtime might support Recursive Read-only (RRO) mounts,
+			// but the underlying kernel might not.
+			if err := checkKernelRROMountSupport(); err != nil {
+				logrus.Warnf("Runtime handler %q supports Recursive Read-only (RRO) mounts, but kernel does not: %v", name, err)
```

Comment:
**To the reviewer:**

An unsupported platform emits:

```console
WARN[2024-04-09 10:25:56.665685980Z] Runtime handler "runc" supports Recursive Read-only (RRO) mounts, but kernel does not: kernel version "5.10.214" does not support recursive read-only mounts: unable to set recursive read-only mount attribute: function not implemented 
WARN[2024-04-09 10:25:56.671210757Z] Runtime handler "crun" supports Recursive Read-only (RRO) mounts, but kernel does not: kernel version "5.10.214" does not support recursive read-only mounts: unable to set recursive read-only mount attribute: function not implemented 
```

Reply:
The output format remains the same even after following the refactor as suggested in the review comment:

- https://github.com/cri-o/cri-o/pull/7962#pullrequestreview-2002056079